### PR TITLE
Add question about rust courses

### PR DESCRIPTION
**Pull request by apiraino**
Incorporating suggestion at [Zulip link](https://rust-lang.zulipchat.com/#narrow/stream/402479-t-community.2Frust-survey/topic/reviewing.20changes/near/395094330).

I think the other points in that Zulip message (w.r.t. to users _desiderata_) are now covered in #243 (possibly also other questions)

r? @Kobzol

### CODE REVIEW

**Comment by Kobzol** [comment]
This seems a bit ad hoc focused on courses. Are we not interested in more kinds of learning materials? E.g.:
`What kind of learning materials have you tried? (select all that apply)`
```
- Books (TRPL, Rust for Rustaceans, …)
- Online exercises (Rustlings, Rust by Example, ...)
- Videos
- Courses
- Something else (fill)
```

**Comment by apiraino** [comment]
fair point. I will reformulate the question+options

File: surveys/2023-annual-survey/questions.md [hunk 1]
```diff
@@ -356,6 +356,12 @@ Type: select one (optional)
 >
 > This question is primarily used to funnel respondents into the more specific questions about the kinds of educational activities they've been a part of.
 
+### If you took a public course, which one was it?
```
Comment (apiraino) [inline surveys/2023-annual-survey/questions.md:359]:
```suggestion
### If you consumed learning material about Rust, what kind of content did you consume?
```

File: surveys/2023-annual-survey/questions.md [hunk 2]
```diff
@@ -356,6 +356,12 @@ Type: select one (optional)
...
+
+Type: free text
+
+> **justification**
+> Justification: getting data here seems helpful for guiding users / recommending public content.
```
Comment (apiraino) [inline surveys/2023-annual-survey/questions.md:364]:
```suggestion
### If you consumed learning material about Rust, which kind of material did you consume?

Type: select all that apply (optional)

- Books (TRPL, Rust for Rustaceans, etc.)
- Online exercises (Rustlings, Rust by Example, etc.)
- Videos
- Online courses, webinars
- Other (please specify)

> **justification**
> Justification: getting data here seems helpful for guiding users / recommending public content.
```

**Review by Kobzol — APPROVED** [review]
Looks good! I suggested "TRPL" as a shortcut, but probably in the final text we should spell it out fully, not everyone might be familiar with the shortcut.

**Pull request merged**